components/dialog/dialog.js

```diff
@@ -201,18 +198,26 @@ class Dialog extends PropertyRequiredMixin(LocalizeCoreElement(AsyncContainerMix
 		);
 	}
 
-	updated(changedProperties) {
+	async updated(changedProperties) {
 		super.updated(changedProperties);
-		if (!changedProperties.has('asyncState')) return;
-		if (this.asyncState === asyncStates.complete) {
-			this.resize();
+
+		if (changedProperties.has('asyncState') && this.asyncState === asyncStates.complete) {
+			await this._handleAsyncChildren();
```

Comment:
This feels like a place where we might want to wrap the new behaviour in a client-side feature flag.